Katello/katello
# Fixes #26516 - Update syspurpose status

See issue details for what's changed specifically. 

**To test**
This requires a new Candlepin: http://koji.katello.org/koji/taskinfo?taskID=187908 this will be in nightly once the pipeline is passing (failures not related to candlepin)

A manifest with SKUs having system purpose attributes can be used. It's also very easy to create a custom product and modify the local candlepin database to add some by hand:

```
insert into cp2_product_attributes(name, value, product_uuid) values ('addons', 'Test Addon1,Test Addon2', 'REPLACEME');
insert into cp2_product_attributes(name, value, product_uuid) values ('role', 'Test Role', 'REPLACEME');
insert into cp2_product_attributes(name, value, product_uuid) values ('usage', 'Development', 'REPLACEME');
```

For some registered content host new or otherwise, clear the syspurpose attributes. The status of the system should say 'Not Specified' after refreshing the page once the Candlepin event has been processed.

Make sure the auto-attach is disabled on the host. Without any subscriptions attached, set one or more of the syspurpose values. After a moment, refreshing the page should show 'Mismatched' as there are no subscriptions providing the specified values.

Now, run auto-attach. The subscription which has the syspurpose attributes should be attached to cover the specified values. After a moment, refreshing the page should show 'Matched' as the syspurpose status.

That's basically it! You may also want to test scoped search from the host list to find your host(s) by syspurpose status: sla_status, usage_status, role_status, addons_status, purpose_status (overall status)

If desired the status can also be refreshed by hand rather than waiting for Candlepin to send us an event (though it is fast):

```
::Host.find(5).subscription_facet.update_purpose_status
```
